src/MagnumPlugins/BasisImageConverter/BasisImageConverter.cpp

```diff
@@ -0,0 +1,216 @@
+/*
+    This file is part of Magnum.
+
+    Copyright © 2010, 2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018, 2019
+              Vladimír Vondruš <mosra@centrum.cz>
+    Copyright © 2019 Jonathan Hale <squareys@googlemail.com>
+
+    Permission is hereby granted, free of charge, to any person obtaining a
+    copy of this software and associated documentation files (the "Software"),
+    to deal in the Software without restriction, including without limitation
+    the rights to use, copy, modify, merge, publish, distribute, sublicense,
+    and/or sell copies of the Software, and to permit persons to whom the
+    Software is furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included
+    in all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+#include "BasisImageConverter.h"
+
+#include <cstring>
+#include <algorithm>
+#include <Corrade/Containers/Array.h>
+#include <Corrade/Containers/ArrayView.h>
+#include <Corrade/Containers/StridedArrayView.h>
+#include <Corrade/Utility/ConfigurationGroup.h>
+#include <Magnum/ImageView.h>
+#include <Magnum/Math/Color.h>
+#include <Magnum/PixelFormat.h>
+
+#include <basisu_enc.h>
+#include <basisu_comp.h>
+
+namespace Magnum { namespace Trade {
+
+BasisImageConverter::BasisImageConverter() = default;
+
+BasisImageConverter::BasisImageConverter(PluginManager::AbstractManager& manager, const std::string& plugin): AbstractImageConverter{manager, plugin} {}
+
+auto BasisImageConverter::doFeatures() const -> Features { return Feature::ConvertData; }
+
+Containers::Array<char> BasisImageConverter::doExportToData(const ImageView2D& image) {
+    /* Check input */
+    if(image.format() != PixelFormat::RGB8Unorm
+        && image.format() != PixelFormat::RGBA8Unorm) {
+        Error() << "Trade::BasisImageConverter::exportToData(): unsupported pixel format" << image.format();
+        return {};
+    }
+
+    /* Magnum specific configuration */
+    const Int threadCount = configuration().value<Int>("num_threads");
+    if(threadCount < 1) {
+        Error() << "Trade::BasisImageConverter::exportToData(): invalid configuration: num_threads is expected to be at least 1, but was:" << threadCount;
+        return {};
+    }
+
+    basisu::basis_compressor_params params;
+
+    const bool multithreading = configuration().value<bool>("multithreading");
+    params.m_multithreading = multithreading;
```

Comment:
What happens if 

```ini
multithreading=false
num_threads=8
```

? Shouldn't `multithreading` be implicit based on the number of threads specified (i.e., enabled if `threads != 1`)?

Reply:
Would make sense, yeah

Reply:
In that case `threads=1` is a good default.